### DISCUSSION TITLE
Batocera Lite/Koriki Alpha for RG35XX Plus and RG35XX H - 20240125

### DISCUSSION THREAD

**Discussion by acmeplus** [rg35xx-cfw/rg35xx-cfw.github.io#79] [Announcements]
### Important: 

* This is a consolidated image for both RG35XX Plus and RG35XX H. You can install and actually swap from one system to another while keeping the same content.
* Because of the consolidation of both device features, this release cannot be installed as an update.
* There are still some performance issues with this release, while there are cores/emulators that are running fine, some others are not there yet. In general the build needs to be updated with optimizations, but there's still work to do.

## Installation
* Flash image to a new sdcard (batocera-h5-rg35xx-plus--20240126.img.gz)
* Insert sdcard into the RG35XX Plus
* Press power a few seconds to boot
* NOTE: The first time the system boots it will take a bit longer while the SDCARD is getting expanded and the initial SHARE folder content gets populated

## Cores/Emulators Status:
* Working: amstrad, mame, atari2600, atari5200, atari7800,  wonderswan color, amiga(uae4arm), c64 (libretro x64), pico8, scummvm (core), colecovision (gearcoleco), vectrex, dosbox, pcengine, pcenginecd, g&w, megacd, nes, gb, snes, gbc, gba, openbor, sg-1000, mastersystem, megadrive, gamegear, spectrum, neogeo, neocd, ngpc, psx, psp, odyssey2, drastic, n64 (.zip with parallel libretro core, z64/n64 with mupen64plus), flycast (standalone and core), msx (fmsx core)
* Ports working (you need your own files): MalditaCastilla, Spelunky, Shovel Knight, VVVVVV, CurseOfIssyos, SpaceCadet Pinball
* Broken: scummvm standalone (unless using an external mouse), x86k (bios issue)

### 2024/01/25:
* Consolidated RG35XX Plus and RG35XX H firmware
* Fixed “not connected” network notification (added missing coreutils timeout)
* Fixed issues with some cores not starting (e.g. GBA). This was indirectly caused by the network “not connected” issue
* Fixed retroachievements not being enabled in retroarch after enabled in ES
* Fixed network SHARE issues (BUILDROOT share should be vivible on your network shares once the console is connected to the network)
* Added boot progress bar. Thanks to Rafapardo (spirax_es) who contributed this needed feature
* Added moonlight. Refer to [batocera moonlight documentation](https://wiki.batocera.org/systems:moonlight) for setup information 
* Added xash3d (Half Life) port. Refer to batocera documentation for setup ([Xash3D half flife forums](https://forum.batocera.org/d/6667-xash3d-half-life))
* Included VVVVVV data as authorized by Terry Cavanagh (see the original repo from @JohnnyonFlame https://github.com/JohnnyonFlame/VVVVVV/releases/tag/v2.4-r1 with)
* Fixed controller mappings for ports (MalditaCastilla, Shovel Knight, Spelunky, etc). Note that you still need to provide your own port content and copy it over to the corresponding port folder in ``roms/ports``

### 2024/01/22:
* Fixed shutdown, reboot. Thanks @xonglebongle (adixal) for the hint
* Switched back to GCC compilation, all cores recompiled
* Fixed flycast standalone freezing before game starts
* Fixed Drastic (NDS) not starting
* Added workaround for Drastic control mapping. Add the file ``es_systems_nds.cfg`` to ``[SHARE]/system/configs/emulationstation/``. To use the stylus keep pressed ``R2`` to use DPAD for stylus and A to "click". 
* Confirmed ADB is working
* Fixed Retroarchievements from EmulationStation (you need to have retroarchievements.org account)
* Fixed scrapers 

### 2024/01/19:
* Fixed SHARE partition automatic expansion for single SDCard configuration
* Change SHARE format to expat
* Added additional card filesystem consistency checks for both internal and external SHARE partitions 
* Fixed SSH credentials, use root/linux as default
* Fixed flycast_libretro (dreamcast, Naomi, atomiswave) and made core selection available from emulationstation. If you applied the 2024/01/16 es_system_dreamcast.cfg workaround, remove it. 
* Flycast standalone is still broken, once you are in dreamcast, Naomi, or atomiswave section, press ``SELECT`` to open the ``VIEW OPTIONS`` menu, then go to ``ADVANCED SYSTEM OPTIONS``, and in ``EMULATOR`` choose ``LIBRETRO: FLYCAST``
* Initial bluetooth work (still not usable via frontend)
* Added some GLES3 cores/emulators: mupen64plus-next and Yabasanshiro (Saturn).
 
### 2024/01/16:
* Fixed audio click noise (not perfect but now very subtle)
* Fixed audio volume controls (volume can be controlled with the keys and is saved during reboots)
* Fixed audio not working on some emulators/cores
* Added brightness controls (level is saved across reboots)
* Added samba support. You can mount the SHARE partition as a network drive

### 2023/12/20:
* Fixed Overclock settings that were not correctly setup. In EmulationStation, change default setting in emulationstation settings
* mupen64 now working
* Added additional cores (gearcoleco, gem, race, vm, bk, gearbox, meteor, quicknes, cannonball, mame2003-xtreme, crocosd, supafaust, etc.). Note that not all cores are mapped in EmulationStation yet, in some cases you will need to open retroarch, and select the core manually to test.

## Things that don't work:
* Charging the console will power it on
* Bluetooth
* ~~Brightness control~~
* ~~Audio has a static click sound when starting/stopping an emulator~~
* Some cores/Standalone emulators:
  * ~~Flycast~~
  * ~~Drastic (requires a minor update)~~
  * Amiga performance is not there yet
* ~~Controls in ES are not mapped initially, just press a button and map them using ES mapping tool~~
* HDMI (not tested)
* Sleep/Supend
* ~~SDCard auto-expansion does not work at this time. You need to use two SDCards at this point~~
* Boot time needs to be improved
* ~~Error 256 when launching roms. If you get that error mount the sdcard on your computer and create a folder system/logs on the last partition of the SDcard (SHARE).~~
* Batocera bezels (not tested)
* Menu language change (graphical themes get updated, but the menu remains in English)
* ~~Drastic works but stylus mapping is not correct~~
* ~~Retroachivements via ES. You can set them up via configuration though,[ check this post for instructions](https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/discussions/69#discussioncomment-8164706)~~

## Things that work:
* Wifi
* Controls
* Many cores/emulators
* EmulationStation
* Simplemenu
* Brigthness control (Menu + Volume keys)
* ADB

## Additional features:
* Switching default launcher (ES/Batocera, Simplemenu/Koriki, or Retroarch):
  * Create a new file in ``[SHARE]/system/customlauncher`` with the text ``simplemenu`` in it.  
  * The next time the system starts will launch simplemenu instead of EmulationStation
  * Possible values for the custom launcher file are: ``simplemenu``, ``emulationstation``, and ``retroarch``
* One or two SDCARDs?
  * Since SDCARD SHARE partition expansion does not work at this time, the best solution is to use a second sdcard for content.
  * If you want to use just one SDCARD, you will need to expand the last partition of the SDCARD (partition 4) to use the maximum space available, not that modifying the partition table of the batocera card may break the internal bootloader, so if the sdcard does not boot after expanding the card, refresh the image and use a second card for content.
* General performance:
  * Expect some cores/emulators to perform better than the original RG35XX, but don't be surprised if some of them perform worse at this point. Cores and emulators have not been optimized yet

[![ko-fi](https://ko-fi.com/img/githubbutton_sm.svg)](https://ko-fi.com/A0A1J951S)

<hr /><em>This discussion was created from the release <a href='https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/releases/tag/'>Batocera Lite/Koriki Alpha for RG35XX Plus and RG35XX H - 20240125</a>.</em>

**Comment by acmeplus** [comment]
Note: there was an error with the image packaging and it does not include the full es_input.cfg that also maps the analogue controls on the RG35XX H. You can map the controls once the console boots up by going to menu (START key) then ``Controller & Bluetooth Settings``, and ``Controller Mapping``. Just press a button and then map all the controls.

Alternatively, after you boot completely the first time, compress and copy this file [es_input.cfg.zip](https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/files/14060434/es_input.cfg.zip)  to ``[SHARE]/system/configs/emulationstation/es_input.cfg``

**Comment by pinkonmyshoe** [comment]
Fantastic update thanks! No issues with Retroachievements anymore and the loading screen is a nice addition. Keep up the hard work it's very appreciated.

**Comment by soaremicheledavid** [comment]
Whats happening to **DS games** ? The only buttons that work in games are the D-pad,  volume up for START and X for PAUSE, how do i configure it right ? Holding down on R2 for the stilus dosent do anything, copied the cfg on SHARE partition, also tried to copy it on the second sd card, nothing happens. Changed the view for single image view before starting the game, nothing changes, i have two small images vertical. And how do you exist the games without reseting the console ? No combo works for closing the game.

**Reply by johnsledget** [reply]
try to delete the system/configs/drastic folder manually and try again

**Reply by charlessouzanetto** [reply]
My DS controls are messed. Like they are double actions. Ex: When I press RT for change screen this activate the start button. When I press the A button this this make the action but too press the right d-pad. When I try to remap the buttons, what is hard because the commands are messed, and turn off the device, the changes are not saved. I dont know if this is corrected in the 20240126 version.

**Reply by james-mccloud** [reply]
me too. i have the same problem

**Reply by acmeplus** [reply]
I'll have a look at this, there were some changes introduced when I upgraded the kernel for RG35XX H and the internal control mappings have changed. It's possible they need to be updated for drastic since it's not using the default configuration.

**Reply by james-mccloud** [reply]
Thank you

**Reply by Yuki1991** [reply]
same here :(

**Reply by acmeplus** [reply]
One question for those testing, have you used the es_systems_nds.cfg or just used it without it. I can get controls to work fine with the image as-is (flashed to a new card, no es_system_nds.cfg added). Granted that there's no stylus, but other than the volume keys duplicated as up/down, the dpad, A/B, start buttons work as expected. 

Update: The menu key (or Menu + X combo) does not work to open drastic menu.

**Reply by soaremicheledavid** [reply]
Removed the file es_system_nds.cfg, deleted drastic folder, after that the buttons work, no stilus, you cant open the drastic menu, but you can close the app now with menu + start .  It seems that file was the problem for me, the buttons are not maped corectly L1 it is select, for example, but works, also there is another problem, the D-PAD, when you want to pres right d-pad, sometimes dosent reconize the input, or if you hold right dpad, after 3 seconds it stops moving, tested in Contra & New Super Mario.

EDIT : Why majority of games lag so much ? On stock OS the same games run perfectly, would performance be improved for DS ?

**Reply by james-mccloud** [reply]
I have tried both and problem persist. Buttons works as you mentioned. 👍🏻

**Comment by KingLJN** [comment]
An issue with the system's localization option seems to occur often and can't be fixed by altering the batocera.conf file or the retroarchcustom.cfg files is the font of the RetroArch notifications. It's set to /usr/share/fonts/dejavu/DejaVuSansMono.ttf, but since it doesn't support CJK characters, I wish to switch to /usr/share/fonts/truetype/noto/NotoSansSC-VF.ttf myself, yet RetroArch launches with some sort of invisible override that requires the DejaVuSans font. This is weird - no such override on the Ozone color theme and menu scales for example. What should I do on my end to alter this, and can this be addressed by a future release?

**Reply by acmeplus** [reply]
You can add Retroarch override variables in batocera.conf in the form of ``global.retroach.variable=value``

**Comment by kthod861** [comment]
Wondering if battery level will come at some point as it is not in the "things that don't work" ?

**Reply by dopazz** [reply]
Battery level is displayed inside RetroArch so the sensor functionality is working, it just needs to be added to EmulationStation.

**Reply by acmeplus** [reply]
Yeah. It’s an issue with emulationstation and the way it reads the battery.

**Reply by acmeplus** [reply]
Issue fixed in 20240126 release: https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/releases/tag/rg35xx_plus_h_batocera_lite_20240126

**Comment by dopazz** [comment]
Is there a way to set the clock other than telnet'ing in and running the DATE command?

**Comment by mldpena** [comment]
Seems like there's a a couple of problems with this version:
- Vsync - On the previous version, vsync is working as expected. But on this version, it seems like it's not working. When running GBA games, (like Super Mario Advance), I can see that the framerate is somewhat choppy, even though it's already hitting 60 fps.
- Retroarch settings - After closing the content via Retroarch menu and updating some settings, when I try to save the configuration, it seems like it's not saving.

All testing done on a fresh install SDCARD.

**Reply by kingzope** [reply]
Same issue I've found with retroarch settings.  Configurations are not saving.

**Reply by kingzope** [reply]
I opened the retroarch and batocera config and made some of the changes I wanted and those stuck.

**Reply by acmeplus** [reply]
Some general settings are preserved but any changes to configuration regarding controls or video aspect etc are overwritten by batocera. Basically any setting that is available via batocera is going to be overwritten. You can override that by forcing the variables in batocera.conf in the form of global.retroarch.variable=value

**Reply by mldpena** [reply]
Hi @acmeplus! I've found out the issue regarding the vsync I've mentioned. It's not about vsync but I think it's similar to what muOS is encountering regarding the shaders. When going back to the previous version, even if I have turned off the shader in ES settings, I'm not encountering any stutters. But in this version, I'm encountering the stutter when shaders are off. The only way to remove the stutter is to enable the shaders (which defaults to sharp-billinear-simple)

**Comment by procent85** [comment]
I bought this console a few days ago. This OS is the most beautiful and has a huge number of options. You're doing a great job. Thank you.

**Comment by npaladin2000** [comment]
Still having issues around RetroAchivements, seems to be an issue with ES trying to index them, it ends up restarting. 

Also, not sure if this is intended, but GameBoy seems to detect GB and GBC, but GBC and SuperGameBoy are only detecting .gbc ROMs.

**Comment by acmeplus** [comment]
https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/releases/tag/rg35xx_plus_h_batocera_lite_20240126

**Reply by dopazz** [reply]
There is no discussion attached to the 20240126 release, but I've found that the EmulationStation battery meter stops updating.  I think it may be related to charging; the battery meter was decreasing but once I plugged it in to charge it no longer updated.  When this occurred the correct battery level was visible in RetroArch.

**Reply by acmeplus** [reply]
Thanks. Will investigate.

**Reply by Kimizyka** [reply]
System does not reboot on version 20240126, is this how it should be?

**Reply by acmeplus** [reply]
No, it shouldn't be like that but is also part of the other issues related to the charger, power, etc. For reboot, just press power again to boot again.

**Reply by Kimizyka** [reply]
Ok, thanks for the answer)

**Comment by johnsledget** [comment]
Testing systems :
Sharp x1 - run but no button map
Othello multivision - works fine
Zx81 - works fine
Supervision - works fine
Apple ii - not sure how to play.it can go to config
Apple iigs - the same as apple ii
Commodore pet - does not play
Commodore vic20 - does not play
Commodore c64 - does not play
Commodore c128 - does not play
Amiga cd32 - when open game it hangs up
Super cassette vission - does not play
Pico8 - works fine
Lowres nx - works fine
Arduboy - does not play
Msdos - works fine but button not map
Msx 2+ - works fine but button not map
Msx turbo r - works fine but i dnt know how to play
Sam coupe -works fine but button not map
Pc 98 - does not work
Videopac+ - does not play

Edited:
Easyrpg - works fine but button not map

**Reply by acmeplus** [reply]
Thanks for the report. I know there are some issues with C64/vice. For MSX try the fmsx core instead. For video pac, make sure you have the right bios files.

**Comment by johnsledget** [comment]
Is there issues in saving a game in NDS?

**Reply by johnsledget** [reply]
This is closed. It can save and load. Files are under /system/configs/drastic/savestates

**Comment by itamar050** [comment]
1. Updating from  20240125 to  20240126 caused all games to be removed (SD was repopulated completely - Does the Jan 26th version support FOTA?
2. As reported previously, issues with RetroAchivements:
* When scraping boxart and logging into RetroAchivements - console freezes
* Seems like in general logging into Retroachivements makes the system hang and eventually reboot
3. Battery meter stops updating

Overall awesome job, love seeing this progress and becomes more polished.
Will there also be an option to map the analogue-joystick by tapping the power button like in the Stock OS?

**Comment by qp3jiukc** [comment]
20240126
The loading screen has this problem.

https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/assets/157653883/4965de48-7282-4c71-8d1d-4ded35390ec9

**Reply by acmeplus** [reply]
Thanks, will be fixed in a future release.

**Comment by soaremicheledavid** [comment]
**BUGS!**

When you want to **suspend the system**, with the power button, it works, but when you press it again to wake up the device, the „configure input„ is there to configure the input again.

The **restart** function from the menu, dosent work, only shutdowns the system.

Unable to activate **retroachievements** : error , that is the message we get, when we want to log into retroachievements.

**Battery indicator** doesn't show the correct percentage of the battery, it gets updated only if you restart the device.

**Reply by npaladin2000** [reply]
I ran into this first item too but it turned out to be an artifact from my old es_input.cfg file: I'm using a seperate SD card for ROMs and the configs.  Replace it with whats on the main SD card and im fine, though it's not sleeping on the press.

**Comment by alojzjakob** [comment]
Hi,

since i finished charging my device the os boot time got really long... takes up to 2 minutes every time i shut it down to boot up again next time i turn it on.
It hangs around 15-20% on progress bar for the most time but eventually starts moving. So whatever is being loaded then at that percentage or a process running at that point might be a clue

**Reply by alojzjakob** [reply]
i have a feeling it is doing a first time boot sequence even if it is not a first boot, since first boot is slow at exactly at this stage on loading bar

**Comment by kingzope** [comment]
I tried pulling the default theme from the file system but was unable to locate it.  Are you able to share it for personal modification?

**Reply by luxmenz** [reply]
I zipped it for u :

https://mega.nz/file/f45ngBgD#clCTkXdgjYJJwQGjmje6n6S2IxyL3NBPB1XVtT9Qn44  (Linux partition)

**Reply by kingzope** [reply]
Thanks so much!

**Reply by luxmenz** [reply]
Theme folder :
/usr/share/emulationstation/themes

**Reply by acmeplus** [reply]
In general themes can live internally under /usr/share/emulationstation/themes, but you can also add themes to [SHARE]/system/configs/emulationstation/themes

**Comment by alojzjakob** [comment]
Where to put bios files?
I tried both bios folder on 2nd card and also on SHARE partition, but bios still missing from emulators etc

**Reply by KingLJN** [reply]
Which emulator cores and which files? See the libretro docs to check if the emulator cores looking for correct names or correct checksums

**Reply by alojzjakob** [reply]
Thank you!
After further examination i saw the bios files were 0B size, so i guess these were corrupted because I took them from stock sd card :)
Found another bios collection on github and it is recognized now!

**Reply by KingLJN** [reply]
They're not "corrupted" per se. They're just 0B files for absolutely no reason at all. The OS's internal rootfs has working BIOS files intended for standalone "Game Room" emulators (Specifically PCSXReArmed). That baffled me to no end.

**Reply by luxmenz** [reply]
I sugest u download "FULL-PACK-BIOS-BATOCERA-V38-MINICAKETV" (search on your favorite search engine)
All bios are present.
Put all bios into bios folder, in SHARE

**Comment by gryll79** [comment]
I tried the Shovel Knight port but when i launch it the Dpad doesnt work, select becomes start, and just other random buttons do random things. Anyone run across this problem?

**Reply by gryll79** [reply]
nevermind i figured it out, i changed the export controller config in the shovelknight.sh from "RG35XX" to "Deeplay-Keys"

**Reply by acmeplus** [reply]
Yeah, don't remember if I updated them, but all the ports should work once you replace that string

**Comment by soaremicheledavid** [comment]
Have made a **batocera lightweight theme** for the RG35XX Plus, if acmeplus wants, it can use it in the image, as an option in themes, reminds me a bit of Garlic Os with a twist. Has only 4,12Mb, different colors options, you just need to copy it in themes folder.
Also have a special bootlogo there, if anyone wants to use it, you need to copy it on first sd card with the OS.

**Download :**

https://github.com/soaremicheledavid/ES-Theme-MINIMA


EDIT :  The theme was re-done, includes more colors and navigation buttons!  Delete any previous theme, do a fresh copy.













Sorry  @acmeplus , had to check twice before uploading, the files was **updated** now!

**Reply by npaladin2000** [reply]
I think you mean whevever the "share" location is (that's the second SD card for those of us using a second SD card) but definitely much appreciated. Most of Batocera's themes are oriented towards larger screens.

Love the gamelist, but the main systems carousel needs some work. Looks messed up.

**Reply by soaremicheledavid** [reply]
Thanks! Was made initially for my other device RG552, wanted to bring it to RG35XX Plus.  I know i need to clean some of the code, but it was done in my launch break. Its nothing that cant be fixed, works as it is, is not perfect, but is close. Feel free to clean my mess!

**Reply by npaladin2000** [reply]
That explains it, the RG552 is a LOT wider than the 35XX, and it's probably hard-coded for that aspect ratio. Not sure if I have a chance to do clean-up (I tried it out during a break too) but if I get a chance I'll look. 

There was also a great minimal theme that was done for either JELOS or AnberELEC that was just a game list with video in the background that would be really good for this.

**Reply by KingLJN** [reply]
I'm not knowledgeable on this but since there are a bevvy of EmulationStation distros for different 4:3 screens, will it be easy or hard to put the existing themes from, say, ArkOS or something?

**Reply by npaladin2000** [reply]
Yeah, I actually like Carbon but I think it needs a taller, more condensed font for this screen. I think I did that on some other device but it was a while ago and I probably don't have it stashed anywhere anymore.

**Reply by KingLJN** [reply]
Color scheme and menu layout look great! However would you consider adding back the button guides at the bottom of the screen? That is, unless you consider this theme is purely for those who are all too familiar with all menu hotkeys in ES of course.

**Reply by acmeplus** [reply]
@soaremicheledavid do you have the theme in a repo that maintain periodically? If that's the case then I can just add it as a package to the distribution, otherwise I can just add it manually to the overlay.

**Reply by soaremicheledavid** [reply]
First of all, the theme has been updated at this moment, I have added some new elements, I recommend that **whoever has already downloaded this theme, delete the old one and copy this one in the theme folder.** 

Second, @acmeplus , this was made initially only for me, but decided to share my work with the comunity, because all the themes on batocera are made for large screen devices, and not optimized for a small screen like the RG35XX Plus.  

Also posted it on here : https://github.com/soaremicheledavid/ES-Theme-MINIMA

**Reply by soaremicheledavid** [reply]
@acmeplus I have submitted my theme to the official batocera website, will be maintain periodically, as long as you keep batocera alive for the RG35XX Plus, just made a special repo for this one : https://github.com/soaremicheledavid/ES-Theme-MINIMA , thank you!

**Reply by acmeplus** [reply]
That's great! That's the best approach so it's accessible online like the rest of the themes.

Eventually I'll try to merge batocera support Into the main repository so it gets updates and is properly maintained.  The main issue with that is that the kernel and bootloaders ar not open source. I have limited time, and while I've managed to get the system to boot with an open source kernel, it requires a lot of work to get it to production level.

**Comment by npaladin2000** [comment]
Anyone else try Lynx emulation? Seems to be failing for me,  getting a black screen.

**Reply by itamar050** [reply]
Yeah, played a couple of games (Batman, Blue Lightning)
BIOS placed in the SHARE/bios folder, Emulator is set to AUTO

Are you experiencing this with a specific game?

**Reply by npaladin2000** [reply]
Naah more than one.  I'll need to double check my BIOS, otherwise my ROM storage got corrupted somehow

**Comment by catpws** [comment]
Im on the latest release and mupen64plus-next has a broken retroarch interface, its this rainbow thing. tried reflashing, checking config files if "gl" is set, all seems ok. other systems work fine aswell. Keep up the good work!

**Comment by soaremicheledavid** [comment]
Can you add a line to the /userdata/system/configs/multimedia_keys.conf or  /etc/triggerhappy/triggers.d/ ?  

Because  multimedia_keys.conf file is missing the screenshot function, dosent save when using commander.

**Comment by pinkonmyshoe** [comment]
Could someone help me figure out how to get Space Cadet pinball working? It's one of my favorite pinball games but I'm not able to figure it out since I'm a bit of a novice. Any guide steps would be appreciated!

**Reply by acmeplus** [reply]
You need to get the original files from the windows game and drop them into the ports/spacecadet folder. I don’t remember which files specifically but I believed I dropped all the .dat files there. 

Note that the same applies to those ports. With the exception of VVVVVV that was authorized to the included there the rest of the ports only contain the necessary tools to run on the rg35xx but not the game data files.

**Reply by pinkonmyshoe** [reply]
Thanks for the reply! I tried that but when opening in the device it would just go to a black screen and back to the games list. I checked the code within the Space Cadet sh file and it seems less fleshed out than the other ones in there. I was able to get VVVVVV, curseofissyos and malditacastilla working using your method above so wondering if I'm missing something else.

**Reply by pinkonmyshoe** [reply]
My ports folder also didn't include a ports/spacecadet folder with the new update whereas the other folders were there already. Maybe something went wrong with my update? Just did a fresh install though.

**Reply by acmeplus** [reply]
I’ll check which files are needed.

**Reply by acmeplus** [reply]
You need at the very least the ``PINBALL.DAT`` and the ``*.WAV`` files. SpaceCadetPinball.sh script needs to be updated to:
```bash
#!/bin/sh

export HOME=/userdata/system/
export SDL_GAMECONTROLLERCONFIG=$(grep "Deeplay" "${HOME}/.config/gamecontroller

cd /userdata/roms/ports/SpaceCadetPinball

SDL_FULLSCREEN=1 SpaceCadetPinball
```

**Reply by pinkonmyshoe** [reply]
Thanks for the help however no luck on my end with the script updated and the files in the 'SpaceCadetPinball' folder. Still starts to black screen and then back to game list. Appreciate your time though.

**Reply by acmeplus** [reply]
There may be logs in system/logs/ that may provide some hints about what may be failing.

**Reply by pinkonmyshoe** [reply]
Ahh yeah found the below in this log file: es_launch_stderr. Not sure how to solve though. 

/bin/sh: line 1: batocera-mouse: command not found
2024-02-01 21:33:47,154 ERROR (emulatorlauncher:550):runCommand /userdata/roms/ports/SpaceCadetPinball.sh: line 4: unexpected EOF while looking for matching `"'

/bin/sh: line 1: batocera-mouse: command not found

**Reply by pinkonmyshoe** [reply]
I got it working thanks to your code on the other ports and chatGPT. Thanks so much for your time and help!
Just had to update the code to be:

```
#!/bin/sh

export HOME=/userdata/system/
export SDL_GAMECONTROLLERCONFIG=$(grep "Deeplay" "${HOME}/.config/gamecontrollerdb.txt")

cd /userdata/roms/ports/SpaceCadetPinball

SDL_FULLSCREEN=1 SpaceCadetPinball
```

**Reply by Boof88** [reply]
Now it's my turn... I had issues with the gamecontrollerdb.txt not existing. I've manually created that file. 

After checking the logs, I have code that looks like the following:

#!/bin/sh

export HOME=/userdata/system/
export SDL_GAMECONTROLLERCONFIG=$(grep "Deeplay" "${HOME}/.config/gamecontrollerdb.txt")

cd /userdata/roms/ports/SpaceCadetPinball

#ESUDO SDL_FULLSCREEN=1 ./SpaceCadetPinball

And checking the system logs now is just:
/bin/sh: 1: batocera-mouse: not found
2024-06-26 22:14:12,407 ERROR (emulatorlauncher:555):runCommand 
/bin/sh: 1: batocera-mouse: not found

Any help please/ :)

**Comment by soaremicheledavid** [comment]
@acmeplus My theme was aproved and its on the batocera website, it can be downloaded from Menu --- Updates & Downloads --- Themes   --- **Minima** (Thats the name of the theme). Will be updated periodically and improved for the RG35XX Plus. 

Also if there are people that are using dual-boot, like me, with Emulation Station & Stock OS, have made a theme also for stock os users, it can be found on my github page.

**Thanks again, none would be possibile if was not for you**, Batocera  works great for this device, also i was been told  that Garlic OS for now its discountinued for the RG35XX Plus, because of GPU & CPU performance issues.

**Reply by albatrosify** [reply]
Where did you hear about Garlic OS beeing discontinued for RG35XX plus? Its literally the reason why I am holding on to this device...

**Reply by npaladin2000** [reply]
It's not discontinued, but it is on hold while the dev regains his health.

**Reply by soaremicheledavid** [reply]
MinUI & Batocera are the only custom firmware for this year, that we can use and enjoy, also the stock os had a nice makeover since 30.01.2024 update. Garlic OS maybe next year if we lucky, because of complications and other problems, only  if someone else takes over the project. Anyway, I don't want to go into details, especially since this github page is dedicated to Batocera, and out of respect for the developer will close this subject.

**Comment by dp152578** [comment]
Any idea why NDS games aren't starting up at all? Games go to a black screen and any button pressed returns me to game list.

**Reply by uzumak11** [reply]
Are you on v40?

**Reply by dp152578** [reply]
v39, didn't realize there was a newer version. OTA update said there was nothing new to install. Do you have a link?

**Reply by uzumak11** [reply]
OTA update doesn't work on v39, or any of the prior releases. Search under the releases here, the build was pushed out on march 6th. NDS works fine on this build.

**Reply by alojzjakob** [reply]
here is the latest release, its for some reason only visible in tags and not under releases
https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/tags
https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/releases/tag/rg35xx_plus_h_batocera_20240306
@acmeplus

**Reply by dp152578** [reply]
So I've updated to v40, but still running into the same issue. Black screen and returns to game library :(

**Comment by acmeplus** [comment]
To all of you following this discussion. As you may have seen, the project has now moved into a new project called [knulli](https://knulli.org) with a new GitHub project [knulli-cfw](https://github.com/knulli-cfw) and there's also a dedicated [discord server](https://discord.gg/HXPS3DAeeB) for discussion. 

If you have been helping out here, let me know if you have a different discord username than the one here and I'll give you access to the early alpha (but note that it's still an alpha and not as featured as the existing v40 20240306 build)
